src/pcre2_jit_compile.c

```diff
@@ -5670,11 +5670,17 @@ if (last)
   chars->last_count++;
 }
 
-static int scan_prefix(compiler_common *common, PCRE2_SPTR cc, fast_forward_char_data *chars, int max_chars, sljit_u32 *rec_count)
+#define SCAN_PREFIX_STACK_END 32
+
+static int scan_prefix(compiler_common *common, PCRE2_SPTR cc, fast_forward_char_data *chars)
 {
-/* Recursive function, which scans prefix literals. */
```

Comment:
It took me an embarrassingly long time to work out what a "prefix literal" meant. I _think_ I get it now: you just want to determine the set of characters which could start the pattern, or return an empty set if there's an unknowable set?

So for example, `(x|yz)` the "prefixes" are `x, y` (not z). And in `(x...)?z` the prefixes are `x, z`.

The old code is completely messing with my head - the flow of control is mega-convoluted to follow. So I should perhaps stop worrying about proving whether the behaviour is unchanged, and instead just review whether the new code implements the prefix concept accurately.